src/proxy/http/profiles.rs

```diff
@@ -357,12 +379,38 @@ pub mod router {
         }
     }
 
-    impl<G, M, R, B> svc::Layer<M> for Layer<G, M, R, B>
+    impl<B, T> rt::Recognize<http::Request<B>> for ConcreteDstRecognize<T>
+    where
+        T: WithAddr + Clone + Eq + Hash,
+    {
+        type Target = T;
+
+        fn recognize(&self, _req: &http::Request<B>) -> Option<Self::Target> {
+            let mut weights = self
+                .dst_overrides
+                .iter()
+                .map(|dst| dst.weight)
+                .filter(|weight| *weight > 0)
+                .peekable();
+            if weights.peek().is_none() {
+                // Weights list is empty.
+                return Some(self.target.clone());
+            }
+            let mut rng = rand::thread_rng();
+            let idx = WeightedIndex::new(weights)
```

Comment:
Is constructing a new `WeightedIndex` expected to be costly? If so, should we move it out of the request path so that it's only rebuilt when routes change?